platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/tipset/TipsetEventCreator.java

```diff
@@ -141,14 +143,24 @@ public TipsetEventCreator(
         cryptography = platformContext.getCryptography();
         antiSelfishnessFactor = Math.max(1.0, eventCreationConfig.antiSelfishnessFactor());
         tipsetMetrics = new TipsetMetrics(platformContext, addressBook);
-        tipsetTracker = new TipsetTracker(time, addressBook);
+        final AncientMode ancientMode = platformContext
+                .getConfiguration()
+                .getConfigData(EventConfig.class)
+                .getAncientMode();
+        tipsetTracker = new TipsetTracker(time, addressBook, ancientMode);
         childlessOtherEventTracker = new ChildlessEventTracker();
         tipsetWeightCalculator = new TipsetWeightCalculator(
                 platformContext, time, addressBook, selfId, tipsetTracker, childlessOtherEventTracker);
         networkSize = addressBook.getSize();
 
         zeroAdvancementWeightLogger = new RateLimitedLogger(logger, time, Duration.ofMinutes(1));
         noParentFoundLogger = new RateLimitedLogger(logger, time, Duration.ofMinutes(1));
+
+        if (ancientMode == AncientMode.BIRTH_ROUND_THRESHOLD) {
+            this.nonAncientEventWindow = NonAncientEventWindow.INITIAL_EVENT_WINDOW_BIRTH_ROUND;
+        } else {
+            this.nonAncientEventWindow = NonAncientEventWindow.INITIAL_EVENT_WINDOW_GENERATION;
+        }
```

Comment:
A method like this might be nice:

```
public class NonAncientEventWindow {
    public static NonAncientEventWindow getInitialWindow(@NonNull final AncientMode ancientMode) {
        // return appropriate window
    } 
}
```

Then, you could make `INITIAL_EVENT_WINDOW_BIRTH_ROUND` and `INITIAL_EVENT_WINDOW_GENERATION` private static objects.

Reply:
added method like the one you suggest.